pkg/controller/nodemaintenance/check_reconcile_with_lease_test.go

```diff
@@ -0,0 +1,281 @@
+package nodemaintenance
+
+import (
+	"context"
+	reflect "reflect"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	coordv1beta1 "k8s.io/api/coordination/v1beta1"
+
+	k8sfakeclient "k8s.io/client-go/kubernetes/fake"
+
+	"k8s.io/client-go/kubernetes/scheme"
+	kubevirtv1alpha1 "kubevirt.io/node-maintenance-operator/pkg/apis/kubevirt/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+
+
+
+var _ = Describe("updateConditionWithLeaseAPISupport", func() {
```

Comment:
Redundant. Please remove.
We get the same coverage by executing `updateCondition` against OCP >= 4.2, which happens for every pull request.  We do not want the burden of maintaining the test twice.

Reply:
not exactly the same test. for this test initDrainer will find that leases are supported. also as a postdondition of successful reconcile it is checked that a lease has been created. also checks other scenario (that it work with an existing lease for instance).